x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/PCG.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs;
+
+import com.carrotsearch.hppc.BitMixer;
+
+/**
+ * Permuted congruential generator style random number generator
+ */
+public final class PCG {
```

Comment:
It would be useful for me to understand the properties of this random number generator that we need that SplittableRandom#nextInt doesn't have.